test/integration/targets/vmware_guest/tasks/cdrom_d1_c1_f0.yml

```diff
@@ -137,6 +137,94 @@
       - "cdrom_vm.failed == false"
       - "cdrom_vm.changed == true"
 
+- name: Create VM with multiple CDROMs
+  vmware_guest:
+    validate_certs: False
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    folder: vm
+    name: CDROM-multi
+    datacenter: "{{ dc1 }}"
+    cluster: "{{ ccr1 }}"
+    resource_pool: Resources
+    guest_id: centos64Guest
+    hardware:
+      memory_mb: 512
+      num_cpus: 1
+      scsi: paravirtual
+    disk:
+    - size_mb: 128
+      type: thin
+      datastore: LocalDS_0
+    cdrom:
+    - controller_type: ide
+      controller_number: 0
+      unit_number: 0
+      type: iso
+      iso_path: "[LocalDS_0] base.iso"
+    - controller_type: ide
+      controller_number: 0
+      unit_number: 1
+      type: client
+    - controller_type: ide
+      controller_number: 1
+      unit_number: 0
+      type: none
+    - controller_type: ide
+      controller_number: 1
+      unit_number: 1
+      type: none
+  register: cdrom_vm
+
+- debug: var=cdrom_vm
+
+- name: assert the VM was created
+  assert:
+    that:
+      - "cdrom_vm.failed == false"
+      - "cdrom_vm.changed == true"
+
+- name: Remove the last 2 CDROMs and update the first 2 for the new VM
+  vmware_guest:
+    validate_certs: False
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    folder: vm
+    name: CDROM-multi
+    datacenter: "{{ dc1 }}"
+    cdrom:
+    - controller_type: ide
+      controller_number: 0
+      unit_number: 0
+      type: client
+    - controller_type: ide
+      controller_number: 0
+      unit_number: 1
+      type: iso
+      iso_path: "[LocalDS_0] base.iso"
+    - controller_type: ide
+      controller_number: 1
+      unit_number: 0
+      type: none
+      state: absent
+    - controller_type: ide
+      controller_number: 1
+      unit_number: 1
+      type: none
+      state: absent
+    state: present
+  register: cdrom_vm
+
+- debug: var=cdrom_vm
+
+- name: assert the VM was changed
+  assert:
+    that:
+      - "cdrom_vm.failed == false"
```

Comment:
The failed is useless, since the playbook will stop before, please use the 'is changed' syntax in your test: `- cdrom_vm is changed`

Reply:
Ok, I deleted all the failed assert in the test file. Thanks.